notebook/notebookapp.py

```diff
@@ -1420,8 +1420,12 @@ def init_webapp(self):
         else:
             # SSL may be missing, so only import it if it's to be used
             import ssl
-            # Disable SSLv3 by default, since its use is discouraged.
-            ssl_options.setdefault('ssl_version', ssl.PROTOCOL_TLSv1)
+            # PROTOCOL_TLS selects the highest ssl/tls protocol version that both the client and
+            # server support. When PROTOCOL_TLS is not available use PROTOCOL_SSLv23
+            ssl_options.setdefault(
+                'ssl_version',
+                getattr('ssl', 'PROTOCOL_TLS', ssl.PROTOCOL_SSLv23)
```

Comment:
Sorry, the first parameter for getattr should be the module object (i.e. `ssl` with no quotes). It's only the attribute name that needs to be a string.

Reply:
you're totally right. sorry about that. I am still trying to have a functional dev environment here.
that is why I am pushing things without tested locally. sorry again T_T